lib/Subscriber.d.ts

```diff
@@ -0,0 +1,22 @@
+import {EventEmitter} from 'events'
+import {SUBSCRIBE, NOTIFY} from './Constants'
+import * as Utils from './Utils'
+
+declare enum SubscriberTerminatedCode {
+  SUBSCRIBE_RESPONSE_TIMEOUT = 0, 
+  SUBSCRIBE_TRANSPORT_ERROR = 1, 
+  SUBSCRIBE_NON_OK_RESPONSE = 2, 

```

Comment:
This enum is not in sync with the corresponding termination code constants used in Subscriber.js, it is missing the value for SUBSCRIBE_BAD_OK_RESPONSE with value 3. Therefore all enum values here beginning with SUBSCRIBE_FAILED_AUTHENTICATION are off by 1, so event listeners for the 'terminated' event will interpret those codes incorrectly. E.g. when they receive the value 5 then according to this enum they will interpret it as "RECEIVE_FINAL_NOTIFY", but in fact value 5 is actually "UNSUBSCRIBE_TIMEOUT".

The enum should have these values:
```
declare enum SubscriberTerminatedCode {
  SUBSCRIBE_RESPONSE_TIMEOUT = 0, 
  SUBSCRIBE_TRANSPORT_ERROR = 1, 
  SUBSCRIBE_NON_OK_RESPONSE = 2,
  SUBSCRIBE_BAD_OK_RESPONSE = 3,
  SUBSCRIBE_FAILED_AUTHENTICATION = 4,
  UNSUBSCRIBE_TIMEOUT = 5,
  RECEIVE_FINAL_NOTIFY = 6,
  RECEIVE_BAD_NOTIFY = 7
}
```

Reply:
thanks stefang42. 
My problem that I don't use typescript. (so the part is unchecked).
If you can provide missed callbacks definition I can add it.
(other possibility wait until I learn the ts definitions ;-) or until the push request will be merged to master)